controllers/volumereplicationgroup_controller.go

```diff
@@ -818,13 +820,14 @@ func (v *VRGInstance) processAsPrimary() (ctrl.Result, error) {
 
 func (v *VRGInstance) reconcileAsPrimary() ctrl.Result {
 	result := ctrl.Result{}
-	if len(v.volSyncPVCs) != 0 {
-		result.Requeue = v.reconcileVolSyncAsPrimary()
-	}
-
+	result.Requeue = v.reconcileVolSyncAsPrimary()
 	v.reconcileVolRepsAsPrimary(&result.Requeue)
 	v.kubeObjectsProtect(&result)
 
+	if v.instance.Spec.PrepareForFinalSync {
+		v.instance.Status.PrepareForFinalSyncComplete = v.volSyncFinalSyncPrepared && v.kubeObjectsFinalSyncPrepared
```

Comment:
@hatfieldbrian did you mean both of these have to be true?
`v.volSyncFinalSyncPrepared && v.kubeObjectsFinalSyncPrepared`
What if kube objects backup is not enabled, will this `kubeObjectsFinalSyncPrepared` be defaulted to true? I couldn't trace it easily. I will try to run e2e tests against your PR to make sure that VolSync has not regressed.

Reply:
As I expected. The Relocation got stuck in the prepare because of the above:
```
 oc get drpc -A -o wide && date
NAMESPACE        NAME            AGE   PREFERREDCLUSTER   FAILOVERCLUSTER   DESIREDSTATE   CURRENTSTATE   PROGRESSION          START TIME             DURATION        PEER READY
busybox-sample   busybox-drpc    45m   mgd1               local-cluster     Relocate       Initiating     PreparingFinalSync   2022-12-05T16:13:29Z                   True

Mon Dec  5 11:24:42 EST 2022
```

Reply:
@BenamarMk, thanks for reviewing and finding this bug!  A fix has been tested and pushed.  Test performed:
```sh
$ kube_object_protection_spec="" vrg_final_sync cluster1
volumereplicationgroup.ramendr.openshift.io/bb configured
Mon Dec  5 15:08:15 PST 2022
Mon Dec  5 15:08:17 PST 2022
volumereplicationgroup.ramendr.openshift.io/bb configured
Mon Dec  5 15:08:17 PST 2022
Mon Dec  5 15:08:18 PST 2022

$ vrg_final_sync cluster1
volumereplicationgroup.ramendr.openshift.io/bb configured
Mon Dec  5 15:08:38 PST 2022
Mon Dec  5 15:08:38 PST 2022
volumereplicationgroup.ramendr.openshift.io/bb configured
Mon Dec  5 15:08:39 PST 2022
Mon Dec  5 15:08:39 PST 2022
```

Reply:
@hatfieldbrian there is still an issue with the `Prepare`.  I think it is here: https://github.com/RamenDR/ramen/blob/db0ed7a44e3e1667184504c204627e63198f14ec/controllers/vrg_kubeobjects.go#L80

If that line is true, then we don't set the `kubeObjectsFinalSyncPrepared` to true when kube object capture is not enabled.

Reply:
@BenamarMk you're right.  I did not test the "NoS3" profile case.  I'm debating the `kubeObjectFinalSyncPrepared` where it is set to `true` when it is complete or not needed vs. its complement `kubeObjectFinalSyncRequired` that is set to `true` when it is needed and not complete.

Reply:
I wonder which takes precedence: `NoS3` or `kubeObjectProtection`?  In other words, if `kubeObjectProtection:` is non-nil, `s3Profiles: [NoS3]`, and `prepareForFinalSync: true`, then is a Kube objects final sync required?  It seems the user is conflicted: they specify `kubeObjectProtection` but no valid s3 store to protect them in.  I suppose I'm leaning toward `s3Profiles: [NoS3]` trumping `kubeObjectProtection`.  @BenamarMK @ShyamsundarR Thoughts?

Reply:
I think we agreed in today's sync-up meeting with @BenamarMk that:
- if `kubeObjectProtection:` is non-nil, `s3Profiles: [NoS3]`, and `prepareForFinalSync: true`, then a Kube objects final sync IS required
- still need to fix bug where `kubeObjectProtection:` is `nil`, `s3Profiles: [NoS3]`, and `prepareForFinalSync: true`, then need to consider `kubeObjects` `prepareForFinalSync` not required/complete

Also I realized that if Kube object protection is enabled, the VRG Kube object needs to be stored in S3 before considering its final sync complete as it contains the pointer to the latest complete backup.